WordPress/openverse
# Centralise frontend network error reporting

## Fixes



Fixes #3468 by @krysal

## Description


I've kept this PR as a draft because:
- it is just a _proposal_ for an idea for how to solve this problem, but is not necessarily the simplest (but I think it's worth it)
- it still needs additional testing and probably some small refactoring, but I don't want to invest time into that until someone else has looked at this and given the general approach/idea a :+1: 

## Testing Instructions

Force an error to happen in a network request on the frontend. For example, append a bogus string to the end of media IDs so that they result in 404s. Confirm that those don't get sent to Plausible or Sentry anymore (try it on `main` as well to see that it goes to Sentry).

I don't know how to force a network error. You may be able to do it using your browser's network debugging tab and explicitly blocking a request.

## Checklist


- [N/A] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
